src/Tasks/Common/Resources/Strings.resx

```diff
@@ -381,8 +381,8 @@ The following are names of parameters or literal values and should not be transl
     <value>NETSDK1065: Cannot find app host for {0}. {0} could be an invalid runtime identifier (RID). For more information about RID, see https://aka.ms/rid-catalog.</value>
     <comment>{StrBegin="NETSDK1065: "}</comment>
   </data>
-  <data name="CannotUseAppHostWithoutRuntimeIdentifier" xml:space="preserve">
-    <value>NETSDK1066: The application host cannot be used without specifying a RuntimeIdentifier. Either specify a RuntimeIdentifier or set UseAppHost to false.</value>
+  <data name="CannotPublishFrameworkDependentWithoutRuntimeIdentifier" xml:space="preserve">
+    <value>NETSDK1066: It is not supported to publish a framework-dependent application without specifying a RuntimeIdentifier.</value>
```

Comment:
Hmm, I'm very confused by this key name and message. You can publish a framework-dependent app without specifying a RID, you just can't get an exe for it without one. This is what happens out of the box with `dotnet new mvc; dotnet publish` and this is documented as a framework-dependent deployment.

Reply:
The problem with calling that "framework dependent" is that, while it **is** dependent on a shared framework being installed, it doesn't care which one.  I personally think there should be a distinction between a "portable" (RID agnostic) deployment and a "framework-dependent" (RID-specific) deployment, but I don't call the shots.

So if we call `dotnet publish` without RID a framework-dependent deployment, then what should `dotnet publish --mode fx-dependent` do?  Right now it tries to use the apphost which requires a RID and errors.

Without this change `dotnet publish --mode fx-dependent-no-exe` does not error, which is fine if we call out it's the default when a RID is not specified.

I'm ok with backing out this commit and having the option text explain that `fx-dependent-no-exe` is the default without a RID, `self-contained` is the default with a RID, and `fx-dependent` requires a RID to use.

Reply:
I think we should just make it so that you get an AppHost whenever you have a RID and don't when you don't. (We can allow /p:UseAppHost=false if you really want to be tied to a RID but not have the exe. This is current self-contained false, but the more I think about it, the more useless that case is so I don't think we need to have a nice CLI spelling for it.)

Reply:
Ok, I'll amend this commit to only change the error text to:

```
A RuntimeIdentifier must be specified to publish a framework-dependent application with an application host.
```

and this will only be displayed for `dotnet publish --mode fx-dependent`.  We'll allow `dotnet publish --mode fx-dependent-no-exe` with or without RID (default for without).

Reply:
OK. Sounds good.

Reply:
I'd like to consider iterating on this a bit more though, but I don't want to block the checkin on it.

> I personally think there should be a distinction between a "portable" (RID agnostic) deployment and a "framework-dependent" (RID-specific) deployment

We'd invalidate a lot of knowledge and docs if we say that "framework-dependent" as a general term means tied to particular OS and CPU. I don't call the shots either, it is something we can discuss with the broader team.

> So if we call dotnet publish without RID a framework-dependent deployment, then what should dotnet publish --mode fx-dependent do? 

I would be in favor of having it just produce a portable app without the apphost:

`dotnet publish --mode fx-dependent` --> portable framework-dependent app
`dotnet publish --mode fx-dependent -r win10-x64` --> framework-dependent win10-x64 app

Reply:
Pushed.

Reply:
Sorry, I pushed up without seeing your comments.  I like that proposal, so let me make that happen.

Reply:
For my proposal, I think it only works if we drop the fx-dependent-no-exe mode. Might be worth talking it through some more with others. Feel free to iterate on this with a separate PR.

Reply:
I'll merge when green and we can further that discussion on the CLI side.

I've made a change to the CLI now whereby `fx-dependent` and `fx-dependent-no-exe` do the right thing without RID specified (effectively mean the same thing).  With RID, the former produces the apphost and the latter does not.  The error message I changed just now only results if you explicitly try to explicitly set `UseAppHost` to true without a RID specified.